gapis/api/vulkan/overdraw.go

```diff
@@ -0,0 +1,1636 @@
+					break
+				}
+				cr := cb.CommandReferences().Get(uint32(k))
+				if cr.Type() == CommandType_cmd_vkCmdBeginRenderPass {
+					lastRenderPassArgs = cb.BufferCommands().
+						VkCmdBeginRenderPass().
+						Get(cr.MapIndex())
+					lastRenderPassIdx = api.SubCmdIdx{
+						uint64(i), uint64(j), uint64(k)}
+				}
+			}
+		}
+	}
+
+	return lastRenderPassArgs, lastRenderPassIdx, nil
+}
+
+type stencilImage struct {
+	handle VkImage
+	format VkFormat
+	width  uint32
+	height uint32
+}
+
+type renderInfo struct {
+	renderPass  VkRenderPass
+	depthIdx    uint32
+	framebuffer VkFramebuffer
+	image       stencilImage
+}
+
+func (s *stencilOverdraw) createNewRenderPassFramebuffer(ctx context.Context,
+	cb CommandBuilder,
+	gs *api.GlobalState,
+	st *State,
+	a arena.Arena,
+	oldRenderPass VkRenderPass,
+	oldFramebuffer VkFramebuffer,
+	alloc func(v ...interface{}) api.AllocResult,
+	addCleanup func(func()),
+	out transform.Writer,
+) (renderInfo, error) {
+	oldRpInfo, ok := st.RenderPasses().Lookup(oldRenderPass)
+	if !ok {
+		return renderInfo{},
+			fmt.Errorf("Invalid renderpass %v",
+				oldRenderPass)
+	}
+
+	oldFbInfo, ok := st.Framebuffers().Lookup(oldFramebuffer)
+	if !ok {
+		return renderInfo{},
+			fmt.Errorf("Invalid framebuffer %v",
+				oldFramebuffer)
+	}
+
+	attachDesc, depthIdx, err :=
+		s.getStencilAttachmentDescription(st, a, oldRpInfo)
+	if err != nil {
+		return renderInfo{}, err
+	}
+
+	width, height := oldFbInfo.Width(), oldFbInfo.Height()
+	device := oldFbInfo.Device()
+	image, err := s.createImage(ctx, cb, st, a, device, attachDesc.Fmt(),
+		width, height, alloc, addCleanup, out)
+	if err != nil {
+		return renderInfo{}, err
+	}
+
+	imageView := s.createImageView(ctx, cb, st, a, device,
+		image.handle, alloc, addCleanup, out)
+
+	renderPass := s.createRenderPass(ctx, cb, st, a, device, oldRpInfo,
+		attachDesc, alloc, addCleanup, out)
+	framebuffer := s.createFramebuffer(ctx, cb, st, a, device, oldFbInfo,
+		renderPass, imageView, alloc, addCleanup, out)
+
+	return renderInfo{renderPass, depthIdx, framebuffer, image}, nil
+}
+
+func (s *stencilOverdraw) getStencilAttachmentDescription(st *State,
+	a arena.Arena,
+	rpInfo RenderPassObjectʳ,
+) (VkAttachmentDescription, uint32, error) {
+
+	depthDesc, idx, err := s.getDepthAttachment(a, rpInfo)
+	if err != nil {
+		return NilVkAttachmentDescription, idx, err
+	}
+
+	// Clone it, but with a stencil-friendly format
+	var stencilDesc VkAttachmentDescription
+	if idx != ^uint32(0) {
+		stencilDesc = depthDesc.Clone(a, api.CloneContext{})
+
+		format, err := depthToStencilFormat(depthDesc.Fmt())
+		if err != nil {
+			return NilVkAttachmentDescription, idx, err
+		}
+		stencilDesc.SetFmt(format)
+		if stencilDesc.FinalLayout() !=
+			VkImageLayout_VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL {
+			stencilDesc.SetFinalLayout(
+				VkImageLayout_VK_IMAGE_LAYOUT_GENERAL)
+		}
+	} else {
+		stencilDesc = MakeVkAttachmentDescription(a)
```

Comment:
So we can store off some extra data on the side of VkCreatePhysicalDevice, and grab the supported depth formats. Then you will be able to look up what the valid/legal depth formats are.

This can be done as  a follow-up.